quantulum3/_lang/en_US/tests/quantities.json

```diff
@@ -1373,5 +1373,15 @@
           "surface": "three million, two hundred & forty"
         }
       ]
+  },

```

Comment:
We probably need way more test cases to check if this (rather big) change in the flow works as expected. Please include one test case for every edge case you have considered (i.e. more than two letters, metric prefix, no metric prefix, ...)

Reply:
Yes, that would be nice, but that would mean that units.json needs to be updated. It misses for example MegaLitre. For the MLitre vs MLitrE vs mLitre cases.

It is weird that some units don't have all prefixes, and some units have more prefixes defined than others. Is it really a good idea to hardcode what prefixes belong to what units? Why don't have all prefixes available to all units?

Reply:
> Why don't have all prefixes available to all units?

Because one of the [criteria for a unit to be listed](https://github.com/nielstron/quantulum3#extension) is for it to be *common* which in our case is defined by having a Wikipedia page (or being redirected to it). This is not the case for all units (some unit-prefix combinations are simply not commonly used).

Reply:
> It misses for example MegaLitre. For the MLitre vs MLitrE vs mLitre cases.

Megalitre is correctly recognized by quantulum. `Mlitre` and the such don't make sense to me but I am easily convinced otherwise by some sensible body of text that uses this expression. Adding `<prefix><surface>` is then trivial by manipulating [the load method](https://github.com/nielstron/quantulum3/blob/dev/quantulum3/load.py#L325)

Reply:
What about `mbar` and `Mbar`?

Reply:
>Because one of the criteria for a unit to be listed ...  is defined by having a Wikipedia page
Yes, but if a unit is included, I expect to have all the common prefixes available. It doesn't make any sense to have units included all with a different subset of prefixes available. 

>Adding <prefix><surface> is then trivial by manipulating the load method
So, why isn't this added? It should be added I think.

>What about mbar and Mbar?
That would be still an interesting test to include to the test suite. I missed you had merged that recently. But than again, I expect this to happen always, not because that is manually added.

Reply:
Please read my previous comments. (Now rewritten for clarity)

Reply:
> This is not the case for all units (Take for example "Deci-tonne").

Yes, because a tonne is a different *kind* of unit. It is just MegaGram. We don't have to support Tonne -> MG conversion, that is clearly a non-goal, but we can support full prefix for units for which this is generally expected, such as SI units. If you don't want to fully support SI units with prefixes, specify that as a non-goal on the README so that people won't have wrong expectations from using this. (But I think a lot of people expect all SI units to just work with all prefixes.)

>Megalitre is correctly recognized by quantulum. Mlitre and the such don't make sense IMO. However, I am easily convinced otherwise by some sensible body of text that uses "MLitre"

Sure, but you have asked me in the past to support such "nonsensical" cases, so that's why I added support for that and wanted to add test cases for that:

>>(what about the unit "Km", we should swap the first letter here for a perfect match. What about "litre", is it useful to check for "litrE"?)

>.... sensible body of text that uses "MLitre" (Please post one here as a comment)

Well apparently Mlitre isn't as nonsensical as it me seems, here you have it:

>>>Secondary sewage from the Hyperion works is now used in a variety of ways to alleviate this. In one application, commissioned in 1997, Memcor processes are used in 15 Mlitre/day CMF/RO system to supply the nearby Mobil and Chevron refineries with feed for their boiler water plants. In an earlier project Memcor supplied an 11.5 Mlitre/day CMF plant as pretreatment for RO to provide water used for injection in a barrier scheme to keep out further sea water.

Source: https://www.climate-policy-watcher.org/water-quality/v-1.html

Reply:
Thanks, I will look into that :)

Reply:
I moved the issue of recognizing units like "Mlitre" to #183

Reply:
> If you don't want to fully support SI units with prefixes, specify that as a non-goal on the README so that people won't have wrong expectations from using this.

The list of expectations for supported units is pretty clear in my opinion. But it may be considered to add options like "all_SI" or "all_Bin" for adding a certain set of prefixes to a unit. I moved this to #184

Reply:
Sounds good! I will probably work on improving the flow for this PR when we I can also get these test cases available (because, yes, the flow is sometimes quite chaotic). Ping me when it gets ready.